frontend/src/app/stocks/containers/Stocks.js

```diff
@@ -29,22 +29,59 @@ export default class Stocks extends React.Component {
   }
 
   getStock(stock){
-    const url = `https://api.iextrading.com/1.0/stock/market/batch?symbols=${stock}&types=quote`;
-    return fetch(url, { method: 'GET' })
+
+    if (stock){
+      console.log(typeof stock)
+      const url = `https://api.iextrading.com/1.0/stock/market/batch?symbols=${stock}&types=quote`;
+      return fetch(url, { method: 'GET' })
+        .then((res) => res.json())
+          .then((res) => {
+            if (Object.entries(res).length === 0){
+              this.setState({
+                error: true,
+                isLoading: false
+              });
+            }
+            else {
+              this.setState({
+                stock: res,
+              });
+              this.getChart(stock);
+            }
+          });
+    }
+    else {
+      alert('Please enter a valid stock')
+    }
+
+
+  }
+
+  buyStock(stock, quantity, price){
+    const url = '/api/market/stock/transact';
+    const headers = {
+      'Authorization: ': 'bearer token'
```

Comment:
`token` in this line should be replaced with the contents of window.localstorage.getItem('jwt')

Reply:
Added to next commit